pkg/resource/resid.go

```diff
@@ -97,10 +97,25 @@ func (n ResId) Namespace() string {
 
 // CopyWithNewPrefix make a new copy from current ResId and append a new prefix
 func (n ResId) CopyWithNewPrefix(p string) ResId {
+	if p != "" && n.prefix != "" {
+		return ResId{gvk: n.gvk, name: n.name, prefix: p + ":" + n.prefix, namespace: n.namespace}
+	}
 	return ResId{gvk: n.gvk, name: n.name, prefix: p + n.prefix, namespace: n.namespace}
 }
 
 // CopyWithNewNamespace make a new copy from current ResId and set a new namespace
 func (n ResId) CopyWithNewNamespace(ns string) ResId {
 	return ResId{gvk: n.gvk, name: n.name, prefix: n.prefix, namespace: ns}
 }
+
+// HasSamePrefix check if two ResIds have the same prefix
+func (n ResId) HasSamePrefix(id ResId) bool {
+	if id.prefix == "" {
+		return true
+	}
```

Comment:
look odd.  

comment needs fixing.  what if n.prefix isn't empty?

maybe use properPrefix instead?

I'm not in love with the name properPrefix - feel free to improve

Reply:
I added a function `prefixList` and used it here.